pinot-segment-local/src/main/java/org/apache/pinot/segment/local/recordtransformer/NullValueTransformer.java

```diff
@@ -41,6 +47,14 @@ public NullValueTransformer(Schema schema) {
         }
       }
     }
+
+    if (tableConfig.getValidationConfig().isAllowNullTimeValue() && timeColumnName != null && schema.getSpecForTimeColumn(timeColumnName) != null) {
```

Comment:
`schema.getSpecForTimeColumn(timeColumnName)` can be expensive because it might need to convert the field type, let's only call it once. We can assert that it is not null because if time column is defined, it must be configured within the schema

Reply:
I see. You mean the logic in `TableConfigUtils.validateValidationConfig`?

Reply:
```suggestion
    if (tableConfig.getValidationConfig().isAllowNullTimeValue() && timeColumnName != null) {
      _dateTimeFieldSpec = schema.getSpecForTimeColumn(timeColumnName);
      Preconditions.checkState(_dateTimeFieldSpec != null, "Failed to find time field: %s from schema: %s", timeColumnName, schema.getSchemaName());
      _defaultTimeValueFormat = new DateTimeFormatSpec(_dateTimeFieldSpec.getFormat());
    } else {
      _dateTimeFieldSpec = null;
      _defaultTimeValueFormat = null;
    }
```